starknet-core/src/serde/num_hex.rs

```diff
@@ -9,21 +10,29 @@ pub mod u64 {
     where
         S: Serializer,
     {
-        serializer.serialize_str(&format!("{value:#x}"))
+        if serializer.is_human_readable() {
+            serializer.serialize_str(&format!("{value:#x}"))
+        } else {
+            serializer.serialize_bytes(&value.to_be_bytes())
```

Comment:
This is actually wasteful. `serde` does not support fixed size array but we know this is actually always 8 bytes. Using a byte slice here forces the serializer to add a length prefix unnecessarily.

Reply:
Changed to serialize into `u64` directly instead.